lib/howl/ui/list_widget.moon

```diff
@@ -0,0 +1,287 @@
+-- Copyright 2012-2015 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+import Scintilla from howl
+import PropertyObject from howl.aux.moon
+import highlight, style, TextWidget, StyledTable from howl.ui
+import Matcher from howl.util
+
+append = table.insert
+
+style.define_default 'list_highlight', color: '#ffffff', underline: true
+
+highlight.define_default 'list_selection', {
+  style: highlight.ROUNDBOX,
+  color: '#888888'
+  alpha: 50
+  outline_alpha: 100
+}
+
+reversed = (list) -> [item for item in *list[#list, 1, -1]]
+
+class ListWidget extends PropertyObject
+  new: (@matcher, opts={}) =>
+    super!
+    @text_widget = TextWidget!
+
+    @opts = moon.copy opts
+    with @opts
+      .filler_text or= '~'
+
+    @_max_height = 10 * @text_widget.row_height * 10
```

Comment:
The formatting of this and the following line makes it clear that the unnecessary multiplications holds some special significance, but in my befuddled state I can't figure out what it is?